client/layout/guided-tours/config-elements.js

```diff
@@ -109,6 +109,12 @@ export class Step extends Component {
 		this.skipIfInvalidContext( nextProps, nextContext );
 	}
 
+	componentWillUnmount() {
+		if ( this.props.isLastStep ) {
+			this.context.quit( { finished: true } );
+		}
+	}
```

Comment:
As per Slack, this may replaced by a more implicit check:

```js
hasStepNoBranching() {
  const { branching, step } = this.context;
  return ! Object.keys( branching[ step ] ).length;
}
```